### PR TITLE
Set macOS deployment target to 10.10 to follow pospec

### PR DESCRIPTION
**Pull request by lucas34**
Link to issue #1441 
Solve macOS deployment target discrepancy when using Carthage or pod.

I updated for all target. Moya, MoyaTest, RxMoya and ReactiveMoya.

### CODE REVIEW

**Comment by BasThomas** [comment]
@SD10 according to #1441: https://github.com/Moya/Moya/issues/1441#issuecomment-341722746

**Comment by lucas34** [comment]
Sure, I removed the merge commit

File: Changelog.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Next
 ### Fixed
 - Fixed a bug with missing Content-Type header when using `.requestJSONEncodable` [#1410](https://github.com/Moya/Moya/pull/1410) by [@Vict0rS](https://github.com/Vict0rS).
+- Fixes carthage OS X not targeting 10.10 [#1444](https://github.com/Moya/Moya/pull/1444) by [@lucas34](https://github.com/lucas34).
```
Comment (sunshinejr) [inline Changelog.md]: This is minor, but can you change `Fixes` to `Fixed`, please? :)

**Comment by lucas34** [comment]
Looks like @BasThomas added the merge commit back. Is it ok ?

**Comment by AndrewSB** [comment]
Yup, it looks great. 

Oh interesting, the merge commits are from the GitHub CTA... I wonder if they show up in the history of `master` once we merged. I guess we're going to find out!

**Comment by ashfurrow** [comment]
Thanks a lot for contributing to Moya! I've invited you to join the Moya GitHub organization – no pressure to accept! If you'd like more information on what that means, check out our [contributor guidelines](https://github.com/Moya/contributors) and feel free to reach out with any questions.

**Comment by AndrewSB** [comment]
Wow, that's incredibly annoying
messages image 254709031

Github adds them as clean merges, complete noise.

**Pull request merged**